src/Symfony/Bridge/PhpUnit/Legacy/CoverageListenerTrait.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bridge\PhpUnit\Legacy;
+
+use PHPUnit\Framework\Test;
+
+/**
+ * PHP 5.3 compatible trait-like shared implementation.
+ *
+ * @author Grégoire Pineau <lyrixx@lyrixx.info>
+ *
+ * @internal
+ */
+class CoverageListenerTrait
+{
+    private $sutFqcnResolver;
+
+    public function __construct(callable $sutFqcnResolver = null)
+    {
+        $this->sutFqcnResolver = $sutFqcnResolver;
+    }
+
+    public function startTest($test)
+    {
+        $annotations = $test->getAnnotations();
+
+        if (isset($annotations['class']['covers']) || isset($annotations['method']['covers'])) {
+            return;
+        }
+
+        $sutFqcn = $this->findSutFqcn($test);
+        if (!$sutFqcn) {
+            return;
```

Comment:
Currently, If I have a scenario:
> Oh no, adding `@covers` annotation is boring, but I want to be strict about coverage and not have side-effect coverage reported, let me use this listener to automate it!

and then, when for some reason src class was not found (even typo in test class name like `Foo123` and `Foo124Test` (4 instead of 3)), one have no nice way to know about it.
Please, at least add a user warning (like we do in tests while using deprecated code)